pika/adapters/base_connection.py

```diff
@@ -78,6 +78,29 @@ def __init__(self,
                                              on_open_error_callback,
                                              on_close_callback)
 
+    def __repr__(self):
+        def get_socket_repr(sock):
```

Comment:
This would probably be a pretty handy function to abstract out to the top level of a connection if it matches the format in the RabbitMQ management plugin.